assets/chezmoi.io/docs/user-guide/password-managers/azure-key-vault.md

```diff
@@ -0,0 +1,25 @@
+# Azure Key Vault
+
+chezmoi includes support for [Azure Key Vault secrets](https://learn.microsoft.com/en-us/azure/key-vault/secrets/about-secrets).
+
+A secret value can be retrieved with the `azureKeyVault` template function, for
+example:
+
+```
+exampleSecret = {{ azureKeyVault "my-secret-name" }}
+```
+
+The Azure Key Vault URL can be specified in the chezmoi's config file with
+`azureKeyVault.vaultUrl.
```

Comment:
This should *probably* be a second optional parameter in the event that there are two different projects that need configuring which use different key vaults. `azureKeyVault.vaultUrl` would be used if otherwise unspecified.

I don’t know what an Azure Key Vault URL looks like, but for the suggestion below, I’m going to assume it’s an HTTPS URL. For the case where there may be multiple key vaults referenced, we may want to make it so that the function accepts either `{{ azureKeyVault "my-secret-name" ["vault-url" | "vault-url-alias"] }}` where the aliases would be defined in the config file under `azureKeyVault.vaultUrls` (plural), which would be a `map[string]string`:

```yaml
azureKeyVault:
  vaultUrls:
    default: https://my-default-key-vault-url/
    sekrit_project: https://my-sekrit-project-vault-url/
```

So `{{ azureKeyVault "my-secret-name" }}` would use the `default` URL and `{{ azureKeyVault "my-secret-name" "sekrit_project" }}` would use the `sekret_project` URL.

Reply:
Thanks for the feedback!

> This should _probably_ be a second optional parameter in the event that there are two different projects that need configuring which use different key vaults. `azureKeyVault.vaultUrl` would be used if otherwise unspecified.
> 
> I don’t know what an Azure Key Vault URL looks like, but for the suggestion below, I’m going to assume it’s an HTTPS URL. For the case where there may be multiple key vaults referenced, we may want to make it so that the function accepts either `{{ azureKeyVault "my-secret-name" ["vault-url" | "vault-url-alias"] }}` where the aliases would be defined in the config file under `azureKeyVault.vaultUrls` (plural), which would be a `map[string]string`:

Initially I was also thinking of supporting multiple key vaults but wanted to keep it simple.
I followed the example of the AWS Secrets Manager implementation which also supports only one profile.
Not sure we need this complexity right now as a key vault can hold many secrets.

The Azure Key Vault URL is indeed an HTTPS endpoint like `https://my-personal-key-vault.vault.azure.net/`.
Where the `hostname` is the Azure resource name and needs to be globally unique.

Reply:
I think that this could be trivially extended to support multiple vaults and simplify the general use. Right now, you have to make sure you modify your `config.$FORMAT` file to include a `azureKeyVault.vaultUrl`. If you add a second parameter `{{ azureKeyVault "my-secret-name" ["vault-url"] }}`, then users don’t have to add that configuration before they can do anything else.

We could make `my-personal-key-vault` (no `https://` at the beginning) transform into `https://my-personal-key-vault.vault.azure.net/`, but that feels too magic to begin with and for something that I don’t use.

Reply:
I’ll leave it up to you as to whether you want to change this to support a second parameter, but I think it would be a good idea (of course, I think that the AWS implementation should *also* be changed, but that’s an entirely different request).